chia/wallet/puzzles/singleton_top_layer.py

```diff
@@ -20,6 +20,160 @@
 MELT_CONDITION = [ConditionOpcode.CREATE_COIN, 0, ESCAPE_VALUE]
 
 
+#
+# An explanation of how this functions from a user's perspective
+#
+# Consider that you have some coin A that you want to create a singleton
+# containing some inner puzzle I from with amount T.  We'll call the Launcher
+# coin, which is created from A "Launcher" and the first iteration of the
+# singleton, called the "Eve" spend, Eve.  When spent, I yields a coin
+# running I' and so on in a singleton specific way described below.
+#
+# The structure of this on the blockchain when done looks like this
+#
+#   ,------------.
+#   | Coin A     |
+#   `------------'
+#         |       
+#  ------------------ Atomic Transaction 1 -----------------
+#         v                                          
+#   .------------.       .-------------------------------.
+#   | Launcher   |------>| Eve Coin Containing Program I |
+#   `------------'       `-------------------------------'
+#                                        |                
+#  -------------------- End Transaction 1 ------------------
+#                                        |                
+#  --------------- (2) Transaction With I ------------------
```

Comment:
This does not necessarily have to be a new transaction

Reply:
Added a comment on that